applications/GeoMechanicsApplication/custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_linear_elastic_dynamic.h

```diff
@@ -0,0 +1,735 @@
+            this->CalculateInitialSecondDerivative(rModelPart, rA, rb);
+            mCopyExternalForceVector = true;
+        }
+
+        // only add dynamics to lhs after calculating intial force vector
+        this->AddDynamicsToLhs(rA, rModelPart);
+
+        // Initialize the linear solver, such that the solver can factorize the matrices already. In case the matrices can be pre-factorized, this step is not
+        // performed during calculation.
+        BaseType::mpLinearSystemSolver->InitializeSolutionStep(rA, dummy_rDx, rb);
+    }
+
+    /**
+     * @brief Function to perform the build of the LHS, mass matrix and damping matrix
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     */
+    void BuildLHS(typename TSchemeType::Pointer pScheme, ModelPart& rModelPart, TSystemMatrixType& rA) override
+    {
+        KRATOS_TRY
+
+        KRATOS_ERROR_IF(!pScheme) << "No scheme provided!" << std::endl;
+
+        this->InitializeDynamicMatrix(mMassMatrix, BaseType::mEquationSystemSize, pScheme, rModelPart);
+        this->InitializeDynamicMatrix(mDampingMatrix, BaseType::mEquationSystemSize, pScheme, rModelPart);
+
+        // Assemble all elements
+        const auto timer = BuiltinTimer();
+
+        // getting the array of the conditions
+        const ElementsArrayType& r_elements = rModelPart.Elements();
+        this->CalculateGlobalMatrices(r_elements, rA, rModelPart);
+
+        const ConditionsArrayType& r_conditions = rModelPart.Conditions();
+        this->CalculateGlobalMatrices(r_conditions, rA, rModelPart);
+
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolverLinearElasticDynamic", BaseType::GetEchoLevel() >= 1)
+            << "Build time: " << timer.ElapsedSeconds() << std::endl;
+
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolverLinearElasticDynamic",
+                       (BaseType::GetEchoLevel() > 2 && rModelPart.GetCommunicator().MyPID() == 0))
+            << "Finished parallel building" << std::endl;
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Builds the RHS and solves the system with an already defined LHS
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     * @param rDx The Unknowns vector
+     * @param rb The RHS vector
+     */
+    void BuildRHSAndSolve(typename TSchemeType::Pointer pScheme,
+                          ModelPart&                    rModelPart,
+                          TSystemMatrixType&            rA,
+                          TSystemVectorType&            rDx,
+                          TSystemVectorType&            rb) override
+    {
+        this->BuildRHS(pScheme, rModelPart, rb);
+
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolverLinearElasticDynamic", BaseType::GetEchoLevel() == 3)
```

Comment:
Don't know if there's a level for, but just in case:
```suggestion
        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolverLinearElasticDynamic", BaseType::GetEchoLevel() >= 3)
```